lighthouse-core/gather/gatherers/seo/font-size.js

```diff
@@ -0,0 +1,176 @@
+function getEffectiveRule(property, node, {
+  inlineStyle,
+  attributesStyle,
+  matchedCSSRules,
+  inherited,
+}) {
+  const cssModel = {
+    styleSheetHeaderForId: id => ({id}),
+  };
+
+  const nodeType = node.nodeType;
+  node.nodeType = () => nodeType;
+  const matchedStyles = new CSSMatchedStyles(
+    cssModel,
+    node,
+    inlineStyle,
+    attributesStyle,
+    matchedCSSRules,
+    null,
+    inherited,
+    null
+  );
+
+  const nodeStyles = matchedStyles.nodeStyles();
+  const matchingRule = nodeStyles.find(style => {
+    const foundProperty = style.allProperties.find(item => item.name === property);
+    return foundProperty &&
+      matchedStyles.propertyState(foundProperty) !== CSSMatchedStyles.PropertyState.Overloaded;
+  });
+
+  return matchingRule;
+}
+
+/**
+ * @param {!Object} driver
+ * @param {!Node} node text node
+ * @returns {!{fontSize: number, textLength: number, node: Node, cssRule: WebInspector.CSSStyleDeclaration}}
+ */
+function getFontSizeInformation(driver, node) {
+  const computedStyles = driver.sendCommand('CSS.getComputedStyleForNode', {nodeId: node.parentId});
+  const matchedRules = driver.sendCommand('CSS.getMatchedStylesForNode', {nodeId: node.parentId});
+
+  return Promise.all([computedStyles, matchedRules])
+    .then(result => {
+      const [{computedStyle}, matchedRules] = result;
+      const fontSizeProperty = computedStyle.find(({name}) => name === FONT_SIZE_PROPERTY_NAME);
+
+      return {
+        fontSize: parseInt(fontSizeProperty.value, 10),
+        textLength: node.nodeValue.trim().length,
+        node: node.parentNode,
+        cssRule: getEffectiveRule(FONT_SIZE_PROPERTY_NAME, node.parentNode, matchedRules),
+      };
+    });
+}
+
+/**
+ * @param {Node} node
+ * @returns {boolean}
+ */
+function isNonEmptyTextNode(node) {
+  return node.nodeType === global.Node.TEXT_NODE && node.nodeValue.trim().length > 0;
+}
+
+class FontSize extends Gatherer {
+  /**
+   * @param {{driver: !Object}} options Run options
+   * @return {!Promise<Array<{fontSize: number, textLength: number, node: Node, cssRule: WebInspector.CSSStyleDeclaration}>>} font-size analysis
+   */
+  afterPass(options) {
+    const stylesheets = new Map();
+    const onStylesheetAdd = sheet => stylesheets.set(sheet.header.styleSheetId, sheet.header);
+    options.driver.on('CSS.styleSheetAdded', onStylesheetAdd);
+
+    const enableDOM = options.driver.sendCommand('DOM.enable');
+    const enableCSS = options.driver.sendCommand('CSS.enable');
+
+    return Promise.all([enableDOM, enableCSS])
+      .then(() => getAllNodesFromBody(options.driver))
+      .then(nodes => nodes.filter(isNonEmptyTextNode))
+      .then(textNodes => Promise.all(
+        textNodes.map(node => getFontSizeInformation(options.driver, node))
+      ))
+      .then(fontSizeInfo => {
+        options.driver.off('CSS.styleSheetAdded', onStylesheetAdd);
+
+        fontSizeInfo
+          .filter(info => info.cssRule && info.cssRule.styleSheetId)
```

Comment:
Just double checked that.`<small>`, `<font size=1>` and friends + nodes that inherit from them (`<small><b>ok</b></small>`) either don't get a `styleSheetID` or even a `cssRule`. 
It's OK though because here we are only connecting CSS rules with URLs of stylesheets they come from and for UA stylesheet we don't have to do that.